src/v/redpanda/admin_server.cc

```diff
@@ -3316,6 +3317,57 @@ void admin_server::register_self_test_routes() {
       });
 }
 
+ss::future<ss::json::json_return_type>
+admin_server::cloud_storage_usage_handler(
```

Comment:
Do we need to expose this via the admin API? It will be exposed via usage already. Also does it matter what node this is run on? Should it be the controller leader?

Reply:
> Do we need to expose this via the admin API? It will be exposed via usage already.

I've exposed on the `debug` endpoint in order to be able to write ducktape tests for this. I think there's no harm in having it there. `debug` routes are not meant for users to consume directly.

> Also does it matter what node this is run on? Should it be the controller leader?

Good question. We can serve the request from any node (leader or follower), but the results may be different depending on the state of the topic table. If the request is served by a follower, there may be pending topic table updates in the controller log (i.e. the topic table is lagging behind), which means that if the request succeeds you can get stale results. The upside is that we can report usage when there's no controller leader available. I think your calling code should decide on whether it want to serve the billing data request from the leader or not (with these trade-offs) in mind.